src/python/gem5/components/boards/kernel_disk_workload.py

```diff
@@ -110,6 +110,17 @@ def _add_disk_to_board(self, disk_image: DiskImageResource) -> None:
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def _finalize_workload(self) -> None:
+        """
+        Finalizes the board setup dependent on the workload.
+        This function connects the IO devices to the board, generates the
+        device tree and expose the location of the dtb to the gem5's workload
+        object. This function should be called after all IO devices are set up
+        (e.g. disk images, PIO devices.)
+        """
+        raise NotImplementedError
+
```

Comment:
I'm not sure if this is necessary as I believe you could change where the disk images are added inside the board itself, though I may be wrong.

Reply:
I have implemented that as you recommended.